Add scripts to build tools.

jeremybennett opened the pull request:
* EXPECTED_BRANCHES: Created.
	* README.md: Add instructions for cloning repos and building tools.
	* build-all.sh: Created.
	* checkout-all.sh: Created.
	* clone-all.sh: Created.

jeremybennett committed:
Add scripts to build tools.

	* EXPECTED_BRANCHES: Created.
	* README.md: Add instructions for cloning repos and building tools.
	* build-all.sh: Created.
	* checkout-all.sh: Created.
	* clone-all.sh: Created.

Signed-off-by: Jeremy Bennett <jeremy.bennett@embecosm.com>

PaoloS02 commented:
Looks good to me!

PaoloS02 commented on build-all.sh:
```diff
@@ -0,0 +1,529 @@
+      exit 1
+    fi
+fi
+#      --with-qemu-src=$TOPDIR/qemu \
```
Commented line I forgot there

jeremybennett replied:
Fixed in second commit.

jeremybennett committed:
Add scripts to run SPEC CPU 2017 and analyse the results.

	* .gitignore: Created.
        * README.md: Add details for runnign SPEC CPU 2017
	* calc-spec-qemu.sh: Created.
	* collate-qemu-times.awk: Created.
	* collate-times.awk: Created.
	* dump-qemu-times.sh: Created.
	* runspec-breakout.awk: Created.
	* runspec-qemu.sh: Created.
	* spec-configs/linux-riscv64-qemu.cfg: Created.
	* spec-configs/peak-flags-qemu.cfg: Created.
	* specbasedata-ref.txt: Created.
	* specbasedata-test.txt: Created.

Signed-off-by: Jeremy Bennett <jeremy.bennett@embecosm.com>

jeremybennett commented:
Merging to allow wider review
